spec/app/ics-020-fungible-token-transfer/README.md

```diff
@@ -637,12 +655,10 @@ function refundTokens(packet: Packet) {
     abortTransactionUnless(false)
   }
 
-  prefix = "{packet.sourcePort}/{packet.sourceChannel}/"
   for token in tokens {
+    onChainDenom = constructOnChainDenom(token.trace, token.denom.base)
     // we are the source if the denomination is not prefixed
```

Comment:
```suggestion
    // we are the source if the denomination is not prefixed
    // Since this is refunding an outgoing packet, we can check if the tokens were originally from the receiver
    // by checking if the tokens were prefixed by our channel end's identifiers.
```